HIP/hip-xxxx.md

```diff
@@ -0,0 +1,88 @@
+
+We have proposed two separate functions to distinguish between fungible and non fungible tokens. One key difference between them is that the FT can set `Fixed` and `Fractional` custom fees, while the NFTs can have `Fixed` and `Royalty`.
+
+## User stories
+
+1. As an EOA I would like to initiate a smart contract transaction that updates a fungible token's custom fees and set a Fixed HBar fee.
+2. As an EOA I would like to initiate a smart contract transaction that updates a fungible token's custom fees and set a Fixed HTS token fee.
+3. As an EOA I would like to initiate a smart contract transaction that updates a fungible token's custom fees and set a Fractional fee.
+4. As an EOA I would like to initiate a smart contract transaction that updates a fungible token's custom fees and set a combination of Fixed and Fractional fees.
+5. As an EOA I would like to initiate a smart contract transaction that updates a non-fungible token's custom fees and set a Fixed HBar fee.
+6. As an EOA I would like to initiate a smart contract transaction that updates a non-fungible token's custom fees and set a Fixed HTS token fee.
+7. As an EOA I would like to initiate a smart contract transaction that updates a non-fungible token's custom fees and set a Royalty fee.
+8. As an EOA I would like to initiate a smart contract transaction that updates a non-fungible token's custom fees and set a combination of Fixed and Royalty fees.
+
+## Specification
+
+The ledger HSCS will utilize the existing Hedera Token Service (HTS) HAPI transaction for `TokenFeeScheduleUpdate` within the System contract logic.
+
+To achieve this the Hedera Token Service (HTS) system contract logic must be updated to expose the necessary custom fee features.
+
+### Hedera Token Service (HTS) system contract
+
+New system contract functions must be added to the `IHederaTokenService` interface to support the updating the custom fees for fungible and non-fungible tokens.
+
+| Hash | Selector | Return | Descriptions |
+| --- | --- | --- | --- |
+| 0xe780c5d3 | function updateFungibleTokenCustomFees(address token,  IHederaTokenService.FixedFee[] memory fixedFees, IHederaTokenService.FractionalFee[] memory fractionalFees) returns (int64 responseCode) | ResponseCode | The response code from the call |
+| 0x01f9eb7d | function updateNonFungibleTokenCustomFees(address token, IHederaTokenService.FixedFee[] memory fixedFees, IHederaTokenService.RoyaltyFee[] memory royaltyFees) returns (int64 responseCode) | ResponseCode | The response code from the call |
+
+## Backwards Compatibility
+
+Backwards compatibility is ensured as no existing features are modified. Similar to HTS system contract this HIP simply exposes HAPI entity functionality.
+
+## Security Implications
+
+The newly added flows will adopt the HAPI authorization logic. In this way the transaction would be successful only if the given contract making the call is set as a `feeScheduleKey` to the token.
```

Comment:
I think add a note in here that says that the `feeScheduleKey` must either be the contract id of the calling contract or a 1 of N threshold key would help clarify what will be needed in order to to complete this call.  Please check that this is what the dispatch to the token service will require.